lib/upload.js

```diff
@@ -219,6 +219,14 @@ class Upload {
     }
   }
 
+  setSize(size) {
```

Comment:
What do we need the setSize method for? We know the size of the stream once the stream has ended. Or is there a situation where this is not enough?

Reply:
@Acconut  A situation where it’s regular ReadStream or Readable stream from nodejs we can’t always expect this behavior for sure. For example in companion we are writing(downloading) in parallel as we are reading(with tus-js), and the reader might hit “null” if a new chunk has not been written since the last chunk was read. But it doesn’t mean that writing has ended. But we can setSize manually in this case when we know for sure that we are done writing.

Reply:
For me this sounds like your implementation has a flaw. A stream not having data to read (i.e. returning null for a read call) and a stream being ended are two different things. That's what the "end" event is for. I have never heard that you can assume a stream is finished just because there is at the moment no data available to consumption. Could it be that there is a misunderstanding by one of us?

Reply:
Oh, I see what you mean. I thought you were referring to the implementation from [here](https://github.com/tus/tus-js-client/pull/126). So you are suggesting that I add the `.end` event to the read stream and remove the `setSize` method? Yeah I can do that

Reply:
The streams from #126 don't have an "end" event but their `read` method returns two values: The actual chunk (which may be null) and a boolean `done` indicating if the stream has ended (see https://github.com/tus/tus-js-client/pull/126/files#diff-94df636fe83ad802ecbd0ade3cd91ff5R42). So you don't listen for the end event but read until the done value turns true.

So, I am not sure what you mean with
>  suggesting that I add the .end event to the read stream

but if you are able to trigger [this code](https://github.com/tus/tus-js-client/pull/126/files#diff-26d16a76eb2641c3108d9b4b67f9cb72R528) when the stream ends and remove the setSize method, I would be happier :)

Reply:
@Acconut I have updated the PR to auto-detect the file size when it is done streaming.